IdentityCore/src/cache/crypto/MSIDAssymetricKeyKeychainGenerator.m

```diff
@@ -240,26 +231,35 @@ - (MSIDAssymetricKeyPair *)generateEphemeralKeyPair:(NSError **)error
 - (MSIDAssymetricKeyPair *)generateKeyPairForKeyDict:(NSDictionary *)attributes
                                                error:(NSError **)error
 {
-    SecKeyRef publicKeyRef = NULL;
-    SecKeyRef privateKeyRef = NULL;
-    OSStatus status = SecKeyGeneratePair((__bridge CFDictionaryRef)attributes, &publicKeyRef, &privateKeyRef);
+    CFErrorRef keyGenerationError = NULL;
+    SecKeyRef privateKeyRef = SecKeyCreateRandomKey((__bridge CFDictionaryRef)attributes, &keyGenerationError);
     
-    if (status != errSecSuccess)
+    if (!privateKeyRef)
+    {
+        NSError *keyError = CFBridgingRelease(keyGenerationError);
+        [self logAndFillError:@"Failed to generate private key." status:(int)keyError.code error:error];
+        return nil;
+    }
+    
+    SecKeyRef publicKeyRef = SecKeyCopyPublicKey(privateKeyRef);
+    if (!publicKeyRef)
     {
-        [self logAndFillError:@"Failed to generate keypair" status:status error:error];
+        [self logAndFillError:@"Failed to copy public key from private key." status:-1 error:error];
```

Comment:
Should we release privateKeyRef here?

Reply:
Yes we should release the private key in case of this error. Addressed.